src/Confluent.Kafka/Producer.cs

```diff
@@ -608,7 +609,7 @@ internal Producer(ProducerBuilder<TKey, TValue> builder)
             var configHandle = SafeConfigHandle.Create();
 
             modifiedConfig.ToList().ForEach((kvp) => {
-                if (kvp.Value == null) throw new ArgumentException($"'{kvp.Key}' configuration parameter must not be null.");
+                if (kvp.Value == null) throw new ArgumentNullException($"'{kvp.Key}' configuration parameter must not be null.");
```

Comment:
could argue this shouldn't be an argument exception, but i think most appropriate.